lib/jxl/dec_reconstruct.cc

```diff
@@ -274,17 +274,21 @@ void DoYCbCrUpsampling(size_t hs, size_t vs, ImageF* plane_in, const Rect& rect,
   // This X,Y range may include more pixels in the output than what we need.
   // Those pixels that fall outside the image boundary are undefined, but it is
   // safe to compute them with MulAdd().
-  size_t y0 = rect.y0() - lf.Padding();
-  size_t y1 = rect.y0() +
-              std::min(frame_rect.y0() + rect.ysize() + lf.Padding(),
-                       frame_dim.ysize_padded) -
-              frame_rect.y0();
-
-  size_t x0 = rect.x0() - lf.Padding();
-  size_t x1 = rect.x0() +
-              std::min(frame_rect.x0() + rect.xsize() + lf.Padding(),
-                       frame_dim.xsize_padded) -
-              frame_rect.x0();
+  const size_t y0 =
+      std::max<ssize_t>(frame_rect.y0() + rect.y0() - yoff - lf.Padding(), 0) +
```

Comment:
I am reasonably sure this is wrong, and should instead be

`rect.y0() + std::max<ssize_t>(frame_rect.y0() - lf.Padding(), 0) - frame_rect.y0();`

Reply:
I simplified that to `rect.y0() - std::min<size_t>(lf.Padding(), frame_rect.y0());`